src/canvas-force-graph.js

```diff
@@ -196,35 +198,38 @@ export default Kapsule({
         beforeCustomLinks.forEach(link => state.linkCanvasObject(link, ctx, state.globalScale));
         ctx.restore();
 
-        // Bundle strokes per unique color/width for performance optimization
-        const linksPerColor = indexBy(defaultPaintLinks, [getColor, getWidth]);
+        // Bundle strokes per unique color/width/dash for performance optimization
+        const linksPerColor = indexBy(defaultPaintLinks, [getColor, getWidth, getLineDash]);
 
         ctx.save();
         Object.entries(linksPerColor).forEach(([color, linksPerWidth]) => {
           const lineColor = !color || color === 'undefined' ? 'rgba(0,0,0,0.15)' : color;
-          Object.entries(linksPerWidth).forEach(([width, links]) => {
+          Object.entries(linksPerWidth).forEach(([width, linesPerLineDash]) => {
             const lineWidth = (width || 1) / state.globalScale + padAmount;
-
-            ctx.beginPath();
-            links.forEach(link => {
-              const start = link.source;
-              const end = link.target;
-              if (!start || !end || !start.hasOwnProperty('x') || !end.hasOwnProperty('x')) return; // skip invalid link
-
-              ctx.moveTo(start.x, start.y);
-
-              const controlPoints = link.__controlPoints;
-
-              if (!controlPoints) { // Straight line
-                ctx.lineTo(end.x, end.y);
-              } else {
-                // Use quadratic curves for regular lines and bezier for loops
-                ctx[controlPoints.length === 2 ? 'quadraticCurveTo' : 'bezierCurveTo'](...controlPoints, end.x, end.y);
-              }
+            Object.entries(linesPerLineDash).forEach(([dashSegments, links]) => {
+              const lineDashSegments = (dashSegments == '' || dashSegments == 'null') ? null : dashSegments.split(',').map(s => parseFloat(s));
+              ctx.beginPath();
+              links.forEach(link => {
+                const start = link.source;
+                const end = link.target;
+                if (!start || !end || !start.hasOwnProperty('x') || !end.hasOwnProperty('x')) return; // skip invalid link
+  
+                ctx.moveTo(start.x, start.y);
+  
+                const controlPoints = link.__controlPoints;
+  
+                if (!controlPoints) { // Straight line
+                  ctx.lineTo(end.x, end.y);
+                } else {
+                  // Use quadratic curves for regular lines and bezier for loops
+                  ctx[controlPoints.length === 2 ? 'quadraticCurveTo' : 'bezierCurveTo'](...controlPoints, end.x, end.y);
+                }
+              });
+              ctx.strokeStyle = lineColor;
+              ctx.lineWidth = lineWidth;
+              lineDashSegments && ctx.setLineDash(lineDashSegments);
```

Comment:
Just realized this always needs to be set, for cases when the first links have are dashed but subsequent ones are not. The property needs to be reset back to solid (`[]`). So:
```
ctx.setLineDash(lineDashSegments || []);
```

Reply:
Done